src/internal/components/focus-portal/index.tsx

```diff
@@ -0,0 +1,132 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef } from 'react';
+
+import Portal from '../portal';
+import FocusDetector from '../focus-detector';
+import { getFirstTabbable, getLastTabbable, getTabbables } from '../../utils/tabbables';
+
+export interface FocusPortalProps {
+  container?: Element;
+  children?: React.ReactNode;
+}
+
+export default function FocusPortal({ container, children }: FocusPortalProps) {
+  const proxyBeforeRef = useRef<HTMLDivElement | null>(null);
```

Comment:
I'd call these refs `childrenBeforeRef` and `childrenAfterRef` to be better clear